astropy/units/structured.py

```diff
@@ -515,3 +515,14 @@ def __ne__(self, other):
             other = other.item()
 
         return self.item() != other
+
+
+def _make_structuredunit_like_dtype(unit: u.UnitBase, dtype: np.dtype) -> StructuredUnit:
```

Comment:
In principle, we could make this automatic by changing https://github.com/astropy/astropy/blob/02009b9f13880b1c219fecfd67bf6b8f9373f8f5/astropy/units/structured.py#L148-L149

One would just do `(unit,) * (len(names) if names is not None else 1)`.

If we keep the function, maybe just `_structured_unit_like_dtype`?

Reply:
I'm not familiar with this section of the code... what other repercussions will this have?

Reply:
`StructuredUnit` would become less strict in requiring `units` and `names` to have the same structure, but rather "broadcast" units over names as needed.
As is, `StructuredUnit(u.one, names=('a', 'b'))` fails.

Reply:
I like the current strictness of ``StructuredUnit``, so I think renaming the function is my preferred solution.

Reply:
Sounds good. We can always change to be more flexible later but hard to go the other way!